### PR TITLE
feat: add banner for promoting GPU enabled machine

### LINKED ISSUE #1582: Promote libkrun podman machine
### Is your enhancement related to a problem? Please describe

As we're giving advice about Podman machine settings when starting a recipe or inference server, we should also promote libkrun Podman machines on macos. 

### Describe the solution you'd like

If the connection is targetting on Podman machine that is not libkrun, we should add a message about the new machine kind

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jeffmaury**
Fixes #1582

### What does this PR do?

Add a banner when creating an inference server or launching a recipe if GPU flag is on but podman machine is not GPU enabled (MacOS only)

### Screenshot / video of UI

https://github.com/user-attachments/assets/f7dca714-56b0-46d3-b117-581b3a5846b7

### What issues does this PR fix or reference?

#1582 

### How to test this PR?

1. Enable the GPU experimental flag
2. Create an inference server on an Applehv machine

### CODE REVIEW

**Comment by feloy** [comment]
When the "Experimental GPU" preference is disabled, I get the message "Non GPU enabled machine
". I would expect to not see it

File: packages/frontend/src/lib/notification/ContainerConnectionWrapper.svelte [hunk 1]
```diff
@@ -26,11 +35,17 @@ $: if (typeof model?.memory === 'number' && containerProviderConnection) {
       connectionInfo = undefined;
       console.error(err);
     });
+  if (fromStore(configuration)?.current?.experimentalGPU && !shouldRecommendGPU(containerProviderConnection)) {
```
Comment (feloy) [inline packages/frontend/src/lib/notification/ContainerConnectionWrapper.svelte:38]: the condition `!shouldRecommendGPU` seems reversed to me

File: packages/frontend/src/lib/notification/ContainerConnectionWrapper.svelte [hunk 2]
```diff
@@ -26,11 +35,17 @@ $: if (typeof model?.memory === 'number' && containerProviderConnection) {
...
+    gpuWarningRequired = true;
+  }
 } else {
   connectionInfo = undefined;
 }
 </script>
 
+{#if !gpuWarningRequired}
```
Comment (feloy) [inline packages/frontend/src/lib/notification/ContainerConnectionWrapper.svelte:46]: same, shouldn't it be the opposite?

**Review by feloy — APPROVED** [review]
LGTM
Tested on Mac, with or without experimental GPU, and with or without libkrun-based virtual machine

**Review by lstocchi — COMMENTED** [review]
Any reason why you don't open the `create new ...` page directly?

**Comment by jeffmaury** [comment]
> Any reason why you don't open the `create new ...` page directly?

Do we have the API ?

**Comment by lstocchi** [comment]
> Do we have the API ?

It seems not, should we update podman desktop? We can also merge this PR as it is and then update it when we have the new API

**Comment by jeffmaury** [comment]
> > Do we have the API ?
> 
> It seems not, should we update podman desktop? We can also merge this PR as it is and then update it when we have the new API

Yes my plan was to have a follow up PR

**Review by lstocchi — APPROVED** [review]
LGTM, works fine!!

**Pull request merged**